spanner/row.go

```diff
@@ -464,14 +464,14 @@ func SelectAll(rows rowIterator, destination interface{}, options ...DecodeOptio
 	var err error
 	return rows.Do(func(row *Row) error {
 		sliceItem := reflect.New(itemType)
-		if isFirstRow && !isPrimitive {
+		if !isPrimitive {
```

Comment:
Do I understand it correctly that we could, instead of the current change, also do the following:
1. Keep the if statements as is.
2. Add a `defer func() { ... set all pointers to nil ...}` to this function.

That would mean that we are re-using the slice of pointers, but they are reset to nil for each new row. Would that be a slightly more efficient solution?

If not, could we otherwise just move the entire declaration of `var pointers` into this function, as I dont' think we are using it outside of the function.

Reply:
I think re-using the slice of pointers will be more optimal here and that was why we initially added this

Reply:
Can we do benchmark test also like here
https://github.com/googleapis/google-cloud-go/pull/9206

To confirm we are not adding regression